code/modules/mob/dead/new_player/login.dm

```diff
@@ -3,8 +3,12 @@
 		return
 
 	if(CONFIG_GET(flag/use_exp_tracking))
-		client.set_exp_from_db()
-		client.set_db_player_flags()
+		client?.set_exp_from_db()
+		client?.set_db_player_flags()
+		if(!client)
```

Comment:
Why not put this check at the start of of this bock, what's the point of calling the two functions if client is already null

Reply:
because these calls can lead to the client becoming null.

read the doc.